File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    queue = deque(enumerate(priorities))

    while queue :
        idx, current_process = queue.popleft()
        
        if not queue :
            answer += 1
            break

        if current_process >= max(queue, key = lambda x : x[1])[1] :
            answer += 1
            if idx == location :
                break
        else :
            queue.append((idx, current_process))
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<Integer> queue = new LinkedList<>();
        for (int p : priorities) {
            queue.offer(p);
        }
        // System.out.println(queue);
        // System.out.println(Collections.max(queue));
        
        int processCnt = 0;
        
        while (!queue.isEmpty()) {
            int value = queue.poll();
            
            // 아무것도 없을 때 처리
            if (queue.isEmpty()) {
                processCnt++;
                answer = processCnt;
                break;
            }
            
            if (value < Collections.max(queue)) {
                queue.offer(value);
                if (location == 0) {
                    location = queue.size()-1;
                } else {
                    location--;
                }
            } else {
                processCnt++;
                if (location == 0) {
                    answer = processCnt;
                    break;
                } else {
                    location--;
                }
            }
        }
        
        return answer;
    }
}